### ISSUE TITLE
Instable on Linux

### ISSUE THREAD

**Issue by oddball** [Mongo2Go/Mongo2Go#47]
The testcase below is rock solid on windows, but fails almost every time on Linux, making impossible to run CI on https://circleci.com/ or https://travis-ci.com/ or any other Linux based CI service.

``` 
using NUnit.Framework;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace MyUnitTest
{
    [TestFixture]
    class TestMongo2go
    {

        [Test]
        public async Task TestStartMany()
        {
            var old_runners = new List<Mongo2Go.MongoDbRunner>();
            Mongo2Go.MongoDbRunner _mongodb;

            for (int i = 0; i < 25; i++)
            {
                string tmpDir = System.IO.Path.GetTempPath();
                _mongodb = Mongo2Go.MongoDbRunner.Start(tmpDir);
                Console.WriteLine("_mongodb.State: " + _mongodb.State);
                bool dbUp = ProbeForMongoDbConnection(_mongodb.ConnectionString, "unittest");
                Console.WriteLine("Connection to " + _mongodb.ConnectionString + " was " + (dbUp ? "successful!" : "NOT successful!"));
                Assert.IsTrue(dbUp, "mongodb is not alive");
                old_runners.Add(_mongodb);
            }

            foreach (Mongo2Go.MongoDbRunner runner in old_runners)
            {
                runner.Dispose();
            }

        }


        private static bool ProbeForMongoDbConnection(string connectionString, string dbName)
        {
            var probeTask =
                    Task.Run(() =>
                    {
                        var isAlive = false;
                        var client = new MongoDB.Driver.MongoClient(connectionString);

                        for (var k = 0; k < 6; k++)
                        {
                            client.GetDatabase(dbName);
                            var server = client.Cluster.Description.Servers.FirstOrDefault();
                            isAlive = (server != null &&
                                   server.HeartbeatException == null &&
                                   server.State == MongoDB.Driver.Core.Servers.ServerState.Connected);
                            if (isAlive)
                            {
                                break;
                            }
                            System.Threading.Thread.Sleep(300);
                        }
                        return isAlive;
                    });
            probeTask.Wait();
            return probeTask.Result;
        }

    }
}
```

**Comment by ash23** [comment]
I can confirm this seems to be related to the target framework (netcoreapp2.0 vs net461) rather than the OS under which Mongo2Go is used.
I.e., also when targeting netcoreapp2.0 under Windows Mongo2Go seems to cause the program to freeze (as soon as the program attempts to access MongoDB via the c# driver). Also the mongod process (launched by Mongo2Go) becomes inaccessible, i.e you cannot even connect to the database using other tools such as NoSql Manager for MongoDB. As soon as the c# application is killed the mongod process again is accessible using other tools.

However when targeting net461 under Windows everything works fine without any problems.

**Comment by perliedman** [comment]
Hi guys, sorry for not commenting on this issue before.

I think there must be something more at play here than just netcore 2.0 and/or Linux. I use Mongo2Go in a CI environment on Linux and run tests through it many times a day and have done so for at least half a year. I run the same tests on Windows (also netcore 2.0) locally without problems. So maybe there's some additional detail that causes the problem you are describing?

I haven't tried @oddball's code on our setup, but could try to find some time to do so later this week and see what happens.

**Comment by JohannesHoppe** [comment]
All the time it worked perfectly on travis-ci (linux), see the build history:
https://travis-ci.org/Mongo2Go/Mongo2Go/builds (last commit from November)

Today I screwed the `dotnet test` up. Does Monog2Go v2.2.2 work for some linux guys here?

**Comment by oddball** [comment]
If I do setup and teardown repeatedly on Linux it fails. Therefor I only I do that once, and then it never fails. The drawback (that I currently live with) is that I dont have a clean db in-between tests.

**Comment by JohannesHoppe** [comment]
Hi @oddball . Many thanks for staying in touch.

Could you also test the very latest version (v2.2.4) on your machine??
On travis it sais now:
https://travis-ci.org/Mongo2Go/Mongo2Go/builds/388746413

> System.TimeoutException: A timeout occured after 30000ms selecting a server using CompositeServerSelector{ Selectors = MongoDB.Driver.MongoClient+AreSessionsSupportedServerSelector, LatencyLimitingServerSelector{ AllowedLatencyRange = 00:00:00.0150000 } }. Client view of cluster state is { ClusterId : "1", ConnectionMode : "Automatic", Type : "Unknown", State : "Disconnected", Servers : [{ ServerId: "{ ClusterId : 1, EndPoint : "Unspecified/localhost:27017" }", EndPoint: "Unspecified/localhost:27017", State: "Disconnected", Type: "Unknown", HeartbeatException: "MongoDB.Driver.MongoConnectionException: An exception occurred while opening a connection to the server. ---> System.Net.Internals.SocketExceptionFactory+ExtendedSocketException: Connection refused 127.0.0.1:27017

**Comment by oddball** [comment]
Hi @JohannesHoppe I am forced to use targetFramework="net461" , so cannot test 2.2.4

**Comment by JohannesHoppe** [comment]
"Connection refused" again. I have no clue why!?

**Comment by oddball** [comment]
I am kind of detached from this issue now, but @JohannesHoppe you start the mongo daemon in your code, correct?! When this issue was important to me, I tried to find the log file of from the mongod, but if I remember correctly it was not there. Is there no way to send a switch to mongod telling it to log everything very verbosely. Perhaps the answer can be found in the log file.

**Comment by bourquep** [comment]
I downgraded Mongo2Go to version 2.2.1 and it works. Tried v2.2.2 and it fails (connection timeout as described above).

So there's something in this diff that broke Mongo2Go on Linux (and in docker containers):

https://github.com/Mongo2Go/Mongo2Go/compare/v2.2.1...v2.2.2

**Comment by bourquep** [comment]
Well, I've been able to develop and run tests all day long, on my Mac as well as on the CI job, using v2.2.1. But now, all of a sudden, tests have started failing to connect to the mongod instance, on my Mac as well as on CI.

Nothing changed, I was just tweaking some tests and it started timing out all of a sudden.

**Comment by georgibakken** [comment]
I am experiencing the same issue with (v2.2.4) in my CI (jenkins). It's running in a Dockerfile based on microsoft/aspnetcore-build:2.0. It runs fine on my Mac.

Running v2.2.1 works fine on both CI and Mac

**Comment by JohannesHoppe** [comment]
fixed in 2.2.6, please test again!

**Comment by georgibakken** [comment]
Version 2.2.6 does the trick, thanks! :)
